packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -133,13 +139,15 @@ class TextSelectionOverlay implements TextSelectionDelegate {
       new OverlayEntry(builder: (BuildContext c) => _buildOverlay(c, _TextSelectionHandlePosition.end)),
     ];
     Overlay.of(context, debugRequiredFor: debugRequiredFor).insertAll(_handles);
+    _controller.forward(from: 0.0);
   }
 
   /// Shows the toolbar by inserting it into the [context]'s overlay.
   void showToolbar() {
     assert(_toolbar == null);
     _toolbar = new OverlayEntry(builder: _buildToolbar);
     Overlay.of(context, debugRequiredFor: debugRequiredFor).insert(_toolbar);
+    _controller.forward(from: 0.0);
```

Comment:
I added a stop() call when the handles are hidden. Though AFAICT it's not harmful if the controller keeps ticking after the overlay goes away.